src/node_main_instance.h

```diff
@@ -14,29 +15,88 @@ namespace node {
 // We may be able to create an abstract class to reuse some of the routines.
 class NodeMainInstance {
  public:
-  NodeMainInstance(const NodeMainInstance&) = delete;
-  NodeMainInstance& operator=(const NodeMainInstance&) = delete;
-  NodeMainInstance(NodeMainInstance&&) = delete;
-  NodeMainInstance& operator=(NodeMainInstance&&) = delete;
+  // An array of indexes that can be used to deserialize data from a V8
+  // snapshot.
+  struct IndexArray {
```

Comment:
Is this type necessary? Couldn't it be an `std::vector` that is `move`d around?
Since we know it's size at code-generation-time it probably could even be an `std:array`.

Reply:
This is modeled after `v8::StartupData`, we could use this to deserialize internal fields later - a difference between this and `std::vector` is that the data is not owned by this structure (just like `v8::StartupData`), which makes it easier to pass the data to V8 without copying.

(Which reminds me..there is a memory leak in the snapshot_builder.cc because the blob data is not freed after it's serialized..)

Reply:
I gave it another try, and looks like we can use a std::vector to manage the ownership after all..I'll switch to that instead, thanks!